frontend/src/store/store.ts

```diff
@@ -1432,68 +1432,35 @@ export default new Vuex.Store<IState>({
+      const {CryptoBlades, SkillToken, Weapons} = state.contracts();
+      if (!CryptoBlades || !SkillToken || !Weapons || !state.defaultAccount) return;
       const chosenElementFee = chosenElement === 100 ? 1 : 2;
       const slippageMultiplier = mintSlippageApproved ? 1.05 : 1;
-
-      if(useStakedSkillOnly) {
-        await CryptoBlades.methods
-          .mintWeaponNUsingStakedSkill(num, chosenElement, eventId)
-          .send({ from: state.defaultAccount, gas: '5000000', gasPrice: getGasPrice(), });
-      }
-      else {
-        await approveFee(
-          CryptoBlades,
-          SkillToken,
-          state.defaultAccount,
-          state.skillRewards,
-          defaultCallOptions(state),
-          defaultCallOptions(state),
-          cryptoBladesMethods => cryptoBladesMethods.getMintWeaponFee(),
-          { feeMultiplier: num * 4 * chosenElementFee * slippageMultiplier, allowInGameOnlyFunds: true }
-        );
-
-        await CryptoBlades.methods.mintWeaponN(num, chosenElement, eventId).send({ from: state.defaultAccount, gas: '5000000', gasPrice: getGasPrice(), });
-      }
-
-      await Promise.all([
-        dispatch('fetchSkillBalance'),
-        dispatch('fetchFightRewardSkill'),
-        dispatch('updateWeaponIds'),
-        dispatch('setupWeaponDurabilities'),
-        dispatch('specialWeaponsManager/fetchShardsSupply')
-      ]);
-    },
-
-    async mintWeapon({ state, dispatch }, { useStakedSkillOnly, chosenElement, eventId = 0, mintSlippageApproved }:
-    { useStakedSkillOnly?: boolean, chosenElement: any, eventId: any, mintSlippageApproved: boolean }) {
-      const { CryptoBlades, SkillToken, Weapons } = state.contracts();
-      if(!CryptoBlades || !SkillToken || !Weapons || !state.defaultAccount) return;
-      const chosenElementFee = chosenElement === 100 ? 1 : 2;
-      const slippageMultiplier = mintSlippageApproved ? 1.05 : 1;
-
-      if(useStakedSkillOnly) {
-        await CryptoBlades.methods
-          .mintWeaponUsingStakedSkill(chosenElement, eventId)
-          .send({ from: state.defaultAccount, gasPrice: getGasPrice(), });
+      const WEAPON_SEED = await CryptoBlades.methods.WEAPON_SEED().call(defaultCallOptions(state));
+
+      if (!await CryptoBlades.methods.hasSeed(WEAPON_SEED, quantity, chosenElement).call(defaultCallOptions(state))) {
+        if (useStakedSkillOnly) {
+          await CryptoBlades.methods
+            .generateWeaponSeedUsingStakedSkill(quantity, chosenElement)
+            .send({from: state.defaultAccount, gasPrice: getGasPrice(),});
+        } else {
+          await approveFee(
+            CryptoBlades,
+            SkillToken,
+            state.defaultAccount,
+            state.skillRewards,
+            defaultCallOptions(state),
+            defaultCallOptions(state),
+            cryptoBladesMethods => cryptoBladesMethods.getMintWeaponFee(),
+            {feeMultiplier: quantity * chosenElementFee * slippageMultiplier, allowInGameOnlyFunds: true}
```

Comment:
does it still work without the magic number? :-D

Reply:
Worked locally :p Needs further checking